### PR TITLE
[Stroage][Sharding] Add a wrapper for StateKvDb.

### PR DESCRIPTION
**Pull request by grao1991**
### Description
Add a wrapper class for StateKvDb.

### Test Plan
Tested using executor benchmark.

### CODE REVIEW

File: storage/aptosdb/src/state_kv_db.rs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+pub struct StateKvDb {
+    state_kv_metadata_db: Arc<DB>,
+    _state_kv_db_shards: [Arc<DB>; NUM_STATE_SHARDS],
+}
+
+impl StateKvDb {
+    // TODO(grao): Support more flexible path to make it easier for people to put different shards
+    // on different disks.
+    pub(crate) fn open<P: AsRef<Path>>(
```
Comment (msmouse) [inline storage/aptosdb/src/state_kv_db.rs:33]: Let's call it `initialize()` to prevent dumb ass like me from thinking it always opens rocksdb

Comment (grao1991) [reply storage/aptosdb/src/state_kv_db.rs]: Changed to `new()`.

File: storage/aptosdb/src/lib.rs [hunk 2]
```diff
@@ -413,19 +403,17 @@ impl AptosDB {
...
+        let ledger_db = Arc::new(ledger_db);
+        let state_kv_db = StateKvDb::new(
+            db_root_path,
+            rocksdb_configs,
+            readonly,
+            Arc::clone(&ledger_db),
+        )?;
+
         if rocksdb_configs.use_state_kv_db {
             info!(state_kv_db_path = state_kv_db_path, "Opened state K/V DB.",);
```
Comment (msmouse) [inline storage/aptosdb/src/lib.rs:418]: * move these logging to `StateKvDb::new()`
* remove `state_kv_db_path`

File: storage/aptosdb/src/state_kv_db.rs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    pub(crate) fn commit(&self, version: Version, state_kv_batch: SchemaBatch) -> Result<()> {
+        state_kv_batch.put::<DbMetadataSchema>(
+            &DbMetadataKey::StateKVCommitProgress,
+            &DbMetadataValue::Version(version),
+        )?;
+
+        // TODO(grao): Support sharding here.
+        self.state_kv_metadata_db.write_schemas(state_kv_batch)
```
Comment (msmouse) [inline storage/aptosdb/src/state_kv_db.rs:87]: call `commit_raw_batch` here?

**Review by msmouse — APPROVED** [review]

File: storage/aptosdb/src/lib.rs (hunk 2 above)
Comment (grao1991) [reply storage/aptosdb/src/lib.rs]: Done

File: storage/aptosdb/src/state_kv_db.rs (hunk 3 above)
Comment (grao1991) [reply storage/aptosdb/src/state_kv_db.rs]: Done

File: storage/aptosdb/src/state_kv_db.rs [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        let _state_kv_db_shards = {
+            if sharding {
+                let mut shard_id: usize = 0;
+                arr![{
+                    let db = Self::open_shard(db_root_path.as_ref(), shard_id as u8, &state_kv_db_config, readonly)?;
+                    shard_id += 1;
+                    Arc::new(db)
+                }; 256]
+            } else {
+                arr![Arc::clone(&state_kv_metadata_db); 256]
```
Comment (areshand) [inline storage/aptosdb/src/state_kv_db.rs:77]: why do we want to use state_kv_metadata_db for kv_db_shards when sharding is not on?

File: storage/aptosdb/src/state_kv_db.rs [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        db_root_path: P,
+        rocksdb_configs: RocksdbConfigs,
+        readonly: bool,
+        ledger_db: Arc<DB>,
+    ) -> Result<Self> {
+        if !rocksdb_configs.use_state_kv_db {
+            info!("State K/V DB is not enabled!");
+            return Ok(Self {
+                state_kv_metadata_db: Arc::clone(&ledger_db),
+                _state_kv_db_shards: arr![Arc::clone(&ledger_db); 256],
```
Comment (areshand) [inline storage/aptosdb/src/state_kv_db.rs:44]: nit: use NUM_STATE_SHARDS to replace 256?

Comment (grao1991) [reply storage/aptosdb/src/state_kv_db.rs:44]: unfortunately the arr! macro only support integer literal ..

**Review by areshand — APPROVED** [review]

**Pull request merged**